sqlglot/dialects/singlestore.py

```diff
@@ -1,5 +1,94 @@
+from sqlglot.dialects.dialect import build_formatted_time
 from sqlglot.dialects.mysql import MySQL
+import typing as t
+from sqlglot import exp, Dialect
+from sqlglot.generator import unsupported_args
+from sqlglot.helper import seq_get
 
 
 class SingleStore(MySQL):
     SUPPORTS_ORDER_BY_ALL = True
+
+    TIME_MAPPING: t.Dict[str, str] = {
+        "AM": "%p",  # Meridian indicator with or without periods
+        "A.M.": "%p",  # Meridian indicator with or without periods
+        "PM": "%p",  # Meridian indicator with or without periods
+        "P.M.": "%p",  # Meridian indicator with or without periods
+        "D": "%u",  # Day of week (1-7)
+        "DD": "%d",  # day of month (1-31)
+        "DY": "%a",  # abbreviated name of day
+        "HH": "%I",  # Hour of day (1-12)
+        "HH12": "%I",  # alias for HH
+        "HH24": "%H",  # Hour of day (0-23)
+        "MI": "%M",  # Minute (0-59)
+        "MM": "%m",  # Month (01-12; January = 01)
```

Comment:
Ditto about padding and using `%-m` instead:

<img width="372" height="69" alt="Screenshot 2025-07-25 at 5 18 34 PM" src="https://github.com/user-attachments/assets/db23607c-1950-4db5-af2e-d09b6612e26e" />

Reply:
The same as days - this is a bug in the SingleStore documentation